### ISSUE TITLE
Extra appname in path..

### ISSUE THREAD

**Issue by justinmchase** [MrJohz/appdirectory#1]
`new AppDirectory('mycoolapp').userData()` on windows produces:

```
C:\Users\awesomeuser\AppData\Local\mycoolapp\mycoolapp
```

But I think it should be:

```
C:\Users\awesomeuser\AppData\Local\mycoolapp
```

Unless the author is explicitly added, I don't think it should be added to the path.

_edited:_ to be more correct

**Comment by justinmchase** [comment]
Actually let me rephrase that, I see that `userData()` actually uses name + version and if version is missing then it will use name twice.

I suppose I should say that if you don't specify the version then it should just use the name. Additionally it would be nice if there was a `appData()` that always used the name but not the version.

**Comment by MrJohz** [comment]
Windows encourages devs to always have both the company name and product name in the path. ([1](https://support2.microsoft.com/default.aspx?scid=kb;en-us;310294#XSLTH3194121123120121120120), [2](https://msdn.microsoft.com/en-us/library/system.windows.forms.application.localuserappdatapath%28v=vs.110%29.aspx))  Where there isn't really a company name, my understanding is that the convention is to use the product name twice.  I'd like to keep to this convention by default (although if you can give a good argument why convention is wrong, or why this isn't convention any more I'll change it), but I do think it should be made possible to explicitly only use the product name once.  My suggestion is that if you pass `false` as a company name, it just won't include the company name portion of the path.  Would that do for you?

**Comment by justinmchase** [comment]
And it seems great to support that; if I provide the company name and appname I would expect them both to be there. Similarly if I provide a version. But as it is there is no good way for me to do what I want, which is to only use appname. A casual perusal of my ~/AppData/Local folder shows me that most apps don't do it the way Microsoft recommends in those links (including Microsoft themselves, e.g. Skype). It may even be good advice but knowing the advice I still don't want to do it. So I have to use AppDirectory like this:

``` JavaScript
var dirs = new AppDirectory('myapp')
var appData = path.dirname(dirs.userData())
```

I haven't checked it yet on other platforms but I may need to wrap it somehow to get what I want, which I really hope I don't have to do. 

The main reasons why I don't want to use the Company name at the root are:
- Sometimes our users need to go into that folder to look at logs or edit files, they typically know our product name not the company name so they end up confused.
- Shallower directory structure is more appealing aesthetically.
- Our existing product (same name, same company name) is being rev'd to v2 but trying to reuse the same folders is more trouble than its worth. We want it to be cleaner, simpler, easier to understand whats in there.

If the default was as I was hoping then you'd end up with a way to do either kind of path, simply by adding more details.

If backwards compatibility was a concern then I may recommend adding a function onto the constructor so I could call it like so:

``` JavaScript
var appdirectory = require('appdirectory')
var appDataDir = appdirectory.appData(appName /*, companyName, version */)
```

Then you wouldn't need to alter the behavior of the constructor for current users.
